subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/RegistryAwareClassLoaderHierarchyHasher.java

```diff
@@ -32,14 +33,8 @@ public RegistryAwareClassLoaderHierarchyHasher(ClassLoaderRegistry registry, Has
     private static Map<ClassLoader, String> collectKnownClassLoaders(ClassLoaderRegistry registry) {
         Map<ClassLoader, String> knownClassLoaders = Maps.newHashMap();
 
-        String javaVmVersion = String.format("%s|%s", System.getProperty("java.vm.name"), System.getProperty("java.vm.vendor"));
-        ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
-        if (systemClassLoader != null) {
-            addClassLoader(knownClassLoaders, systemClassLoader, "system-app" + javaVmVersion);
-            addClassLoader(knownClassLoaders, systemClassLoader.getParent(), "system-ext" + javaVmVersion);
-        }
-
         String gradleVersion = GradleVersion.current().getVersion();
+        addClassLoader(knownClassLoaders, null, "bootstrap");
```

Comment:
So we are having `null` as a key in the map of known classloaders now? I suppose this works though it is unexpected. Can we add a comment somewhere why we do this?